### PR TITLE
260 update sector schema to allow simplified specification

### PR DESCRIPTION
**Pull request by jsolbrig**
# Reviewer Checklist
https://github.com/NRLMMD-GEOIPS/.github/blob/main/.github/review-template.md

# Related Issues
fixes NRLMMD-GEOIPS/geoips#262
fixes NRLMMD-GEOIPS/geoips#260
fixes NRLMMD-GEOIPS/geoips#110
fixes NRLMMD-GEOIPS/geoips#111
fixes NRLMMD-GEOIPS/geoips#166

# Testing Instructions

# Summary

# Output

### CODE REVIEW

**Comment by jsolbrig** [comment]
For now, in this PR, I will not implement `center` and `corners` families. Instead, I have just simplified the `area_definition` based sectors by making them use the `center` property rather than `area_extent`. This still results in a fully defined `AreaDefinition` without needing to compute `area_extent`.

For `center` and `corners`, please see issues #284 and #301.

File: geoips/interfaces/base.py [hunk 1]
```diff
@@ -170,18 +170,46 @@ def validate(self, plugin, validator_id=None):
...
                 f"\nin interface '{plugin['interface']}'"
-            )
+            ) from err
+
+        # This turned out to be too big of a change for now.
+        # We should consider how to implement something like this while still being able
+        # to test the preceeding error in our unit tests. Currently, the error ourput by
+        # `valicator.validate(plugin)` is being used for testing in the unit tests.
+        #
+        # See issue #303
```
Comment (jsolbrig) [inline geoips/interfaces/base.py:199]: I created issue #303 to deal with this, but would like to leave this as an example for use in #303.

File: geoips/interfaces/yaml_based/sectors.py [hunk 2]
```diff
@@ -12,13 +12,91 @@
...
-from geoips.interfaces.base import BaseYamlInterface
+from geoips.interfaces.base import BaseYamlPlugin, BaseYamlInterface
+from pyresample import load_area
+from geoips.image_utils.mpl_utils import create_figure_and_main_ax_and_mapobj
+from cartopy import feature as cfeature
+from pyresample import create_area_def
+
+
+# Commenting these out for PR #260
+# Will work on this again after the 2023 workshop
```
Comment (jsolbrig) [inline geoips/interfaces/yaml_based/sectors.py:23]: I added #301 to work on this. I'd like to leave these here as examples for #301.

**Comment by jsolbrig** [comment]
@mindyls This is ready for review. It passes `full_test.sh`.

There are two commented code sections that I would like to leave in place. They will be used in #301 and #303.

**Review by mindyls — CHANGES_REQUESTED** [review]
I appear to be getting a validation error on this branch: 

```
        try:
            validator = self.validators[validator_id]
        except KeyError as err:
>           raise ValidationError(
                f"No validator found for '{validator_id}'"
                f"\nfrom plugin '{plugin['name']}'"
                f"\nin interface '{plugin['interface']}'"
            ) from err
E           jsonschema.exceptions.ValidationError: No validator found for 'sectors.area_definition_static'
E           from plugin 'CONUS'
E           in interface 'sectors'

geoips/interfaces/base.py:188: ValidationError
======================================================================= short test summary info =======================================================================
FAILED tests/test_pytest/test_plugin_schema/test_yaml_schema.py::test_is_plugin_valid[sectors.area_definition_static-plugin28] - jsonschema.exceptions.ValidationError: No validator found for 'sectors.area_definition_static'
=================================================================== 1 failed, 1317 passed in 29.85s ===================================================================
```

File: geoips/schema/sectors/static.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+required:
+  - spec
+  - metadata
+# Check to be sure that the family is a known family
+# The first group defines which spec schema to use
+# The second group defines which metadata schema to use
+# All of these must exist in the list of if-statements below
+properties:
+  family:
+    pattern: "^((area_definition)|(center))_((static)|(pyrocb)|(atmosriver)|(stitched)|(tc)|(volcano))$"
```
[the remark this answers is not in the document]
Comment (jsolbrig) [reply geoips/schema/sectors/static.yaml:17]: Yeah, I was noticing that we had something like six schema already and, when I went to add `center_` I was going to need to add six more! It was only going to keep growing.

**Review by mindyls — APPROVED** [review]
pytest error is fixed now.

I think I am ok merging with the schema as is (https://github.com/NRLMMD-GEOIPS/geoips/pull/279#discussion_r1307435833), and we can reassess at a later date.

File: geoips/interfaces/base.py [hunk 4]
```diff
@@ -438,6 +412,64 @@ def __init__(self):
             load_all_yaml_plugins()
         )
 
+    @classmethod
```
[the remark this answers is not in the document]
Comment (jsolbrig) [reply geoips/interfaces/base.py:415]: Right, nothing changed. Making it a classmethod, though, made it possible to create things like `SectorPluginBase` without circular imports.

**Review by christopher-selman — APPROVED** [review]

File: geoips/schema/sectors/static.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (mindyls) [reply geoips/schema/sectors/static.yaml:17]: https://github.com/NRLMMD-GEOIPS/geoips/issues/305

Created that issue. Let's not address this now.  We'll just merge this for now.

**Review by mindyls — CHANGES_REQUESTED** [review]
Add release notes please.  I'd like to see your summary of these changes...

sectors/static/*.yaml is fine for the YAML updates (don't have to note every single file)

**Review by mindyls — APPROVED** [review]
Release notes added, tests pass.

**Pull request merged**